src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -0,0 +1,231 @@
+            6. IOR should complete without any error.
+            7. Drive status LED should be solid Amber ON indicating it’s faulty.
+            8. Replace the same drive back.
+            9. Drive status LED should be off indicating good device is plugged-in.
+           10. Reset and cleanup.
+        Args:
+            oclass (str, optional): object class (eg: RP_2G8, S1,etc). Defaults to None
+        """
+        # 1.
+        self.log_step("Bring up server and reset led of all devices.")
+        if oclass is None:
+            oclass = self.ior_cmd.dfs_oclass.value
+        device_info = get_storage_query_device_info(self, self.dmg)
+        devices = [device['uuid'] for device in device_info]
+        self.log.info("==device_info= %s", devices)
+        test_dev = devices[0]
+        for device in devices:
+            self.run_vmd_led_identify(device, reset=True)
+
+        # 2.
+        self.log_step("Check drive status led state.")
+        err_dev = []
+        for device in devices:
+            if not self.verify_dev_led_state(device):
+                err_dev.append(device)
+        if err_dev:
+            self.fail("#Device {} not in expected NORMAL, OFF state".format(err_dev))
+
+        # 3.
+        self.log_step("Create the pool and container with RF and start IOR")
+        self.add_pool(connect=False)
+        self.add_container(self.pool)
+
+        job_manager = get_job_manager(self, subprocess=None, timeout=120)
+        thread_queue = Queue()
+
+        ior_test_seq = self.params.get("ior_test_sequence", "/run/ior/iorflags/*")[0]
+        thread = []
+        kwargs = {"thread_queue": thread_queue, "job_id": 1}
+        ior_kwargs = {
+            "test": self,
+            "manager": job_manager,
+            "log": "ior_thread_write_pool_test.log",
+            "hosts": self.hostlist_clients,
+            "path": self.workdir,
+            "slots": None,
+            "group": self.server_group,
+            "pool": self.pool,
+            "container": self.container,
+            "processes": self.params.get("np", "/run/ior/client_processes/*"),
+            "ppn": self.params.get("ppn", "/run/ior/client_processes/*"),
+            "intercept": None,
+            "plugin_path": None,
+            "dfuse": None,
+            "display_space": True,
+            "fail_on_warning": False,
+            "namespace": "/run/ior/*",
+            "ior_params": {
+                "oclass": oclass,
+                "flags": self.ior_w_flags,
+                "transfer_size": ior_test_seq[0],
+                "block_size": ior_test_seq[1]
+            }
+        }
+        kwargs.update(ior_kwargs)
+        thread.append(threading.Thread(target=thread_run_ior, kwargs=kwargs))
+
+        # Launch the IOR thread
+        self.log.info("Start IOR thread : %s", thread)
+        thread[0].start()
+
+        # 4.
+        self.log_step("dmg storage set nvme-faulty and check led state")
```

Comment:
Same comment about adding more detail for the person triaging the test failure:
```suggestion
        self.log_step("Marking the {} device as faulty and verifying it is 'EVICTED' and its LED is 'ON'".format(test_dev))
```